packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -249,10 +234,48 @@ export const Product = ({
       <h3 className="billing-title">
         <span>Billing Information</span>
       </h3>
+
+      <TransitionGroup component={null}>
+        {createTokenError.error && (
+          <CSSTransition classNames="animate-fade" timeout={150}>
+            <ErrorMessage>
+              <p data-testid="error-payment-submission">
+                {getErrorMessage(createTokenError.type)}
+              </p>
+            </ErrorMessage>
+          </CSSTransition>
+        )}
+      </TransitionGroup>
+
+      <TransitionGroup component={null}>
+        {createSubscriptionStatus.error !== null && (
+          <CSSTransition classNames="animate-fade" timeout={150}>
+            {createSubscriptionStatus.error.code === 'card_declined' ? (
+              <ErrorMessage>
+                <p data-testid="error-card-declined">
+                  {getErrorMessage('card_error')}
+                </p>
+              </ErrorMessage>
+            ) : (
+              <DialogMessage
+                className="dialog-error"
+                onDismiss={resetCreateSubscriptionError}
+              >
+                <h4 data-testid="error-subscription-failed">
+                  Subscription failed
+                </h4>
+                <p>{createSubscriptionStatus.error.message}</p>
+              </DialogMessage>
```

Comment:
I removed this from being its own component (`CreateSubscriptionErrorDialog`) because `onDismiss` is required for `DialogMessage` but not for `ErrorMessage`. TypeScript complained when I made it optional since it could be undefined.

Reply:
Should the `<DialogMessage>` be a child of the `<TransitionGroup>` here? We could extract the conditional out into a variable like `isCVCError` up next to `inProgress`. Then, use `isCVCError && <ErrorMessage>` and `!isCVCError && <DialogMessage>`

```javascript
const isCVCError =
  createSubscriptionStatus.error !== null &&
  ( createSubscriptionStatus.error.code === 'card_declined' ||
    createSubscriptionStatus.error.code === 'incorrect_cvc');
```